authoring-extension/src/helper/common.ts

```diff
@@ -260,3 +263,19 @@ export function isMarkdownFileCheck(editor: vscode.TextEditor, languageId: boole
 export enum LogType {
     Telemetry, Trace,
 }
+
+/**
+ * Create timestamp
+ */
+export function generateTimestamp() {
```

Comment:
You may be able to get the same results here with less code with something like this:

```
export function generateTimestamp() {
    const date = new Date(Date.now());
    msDateValue = date.toLocaleDateString('en-us');
    msTimeValue = date.toLocaleTimeString({}, { hour12: false });
}
```

It also might be worth it to make sure that these dates are using the UTC time zone so that all dates are consistent.

Finally, you could return the values instead of setting module variables, and just use the output directly, like so:

```
...
return {
    msDateValue: date.toLocaleDateString('en-us'),
    msTimeValue: date.toLocaleTimeString({}, { hour12: false })
};

// in quick-pick-menu-controller
...
catch (error) {
    const { msTimeValue } = common.generateTimestamp(); // You could also skip the assignment and just use the returned property
    output.appendLine(msTimeValue + ...)
}
```

Reply:
Great idea, updated